packages/tx/src/util.ts

```diff
@@ -116,6 +129,98 @@ export class AccessLists {
   }
 }
 
+export class AuthorizationLists {
+  public static getAuthorizationListData(
+    authorizationList: AuthorizationListBytes | AuthorizationList
+  ) {
+    let AuthorizationListJSON
+    let bufferAuthorizationList
+    if (isAuthorizationList(authorizationList)) {
+      AuthorizationListJSON = authorizationList
+      const newAuthorizationList: AuthorizationListBytes = []
+      const jsonItems = ['chainId', 'address', 'nonce', 'yParity', 'r', 's']
+      for (let i = 0; i < authorizationList.length; i++) {
+        const item: AuthorizationListItem = authorizationList[i]
+        for (const key of jsonItems) {
+          // @ts-ignore TODO why does TsScript fail here?
+          if (item[key] === undefined) {
+            throw new Error(`EIP-7702 authorization list invalid: ${key} is not defined`)
+          }
+        }
+        const chainId = hexToBytes(item.chainId)
+        const addressBytes = hexToBytes(item.address)
+        const nonceList = []
+        for (let j = 0; j < item.nonce.length; j++) {
+          nonceList.push(hexToBytes(item.nonce[j]))
+        }
+        const yParity = hexToBytes(item.yParity)
+        const r = hexToBytes(item.r)
+        const s = hexToBytes(item.s)
+
+        newAuthorizationList.push([chainId, addressBytes, nonceList, yParity, r, s])
+      }
+      bufferAuthorizationList = newAuthorizationList
+    } else {
+      bufferAuthorizationList = authorizationList ?? []
+      // build the JSON
+      const json: AuthorizationList = []
+      for (let i = 0; i < bufferAuthorizationList.length; i++) {
+        const data = bufferAuthorizationList[i]
+        const chainId = bytesToHex(data[0])
+        const address = bytesToHex(data[1])
+        const nonces = data[2]
+        const nonceList: PrefixedHexString[] = []
+        for (let j = 0; j < nonces.length; j++) {
+          nonceList.push(bytesToHex(nonces[j]))
+        }
+        const yParity = bytesToHex(data[3])
+        const r = bytesToHex(data[4])
+        const s = bytesToHex(data[5])
+        const jsonItem: AuthorizationListItem = {
+          chainId,
+          address,
+          nonce: nonceList,
+          yParity,
+          r,
+          s,
+        }
+        json.push(jsonItem)
+      }
+      AuthorizationListJSON = json
+    }
+
+    return {
+      AuthorizationListJSON,
+      authorizationList: bufferAuthorizationList,
+    }
+  }
+
+  public static verifyAuthorizationList(authorizationList: AuthorizationListBytes) {
+    for (let key = 0; key < authorizationList.length; key++) {
```

Comment:
The EIP says `If any of the above steps fail, immediately stop processing that tuple and continue to the next tuple in the list. It will In the case of multiple tuples for the same authority, set the code specified by the address in the first occurrence.` about validating each tuple in the authorization list.  It doesn't indicate we should throw an error but just proceed to the next tuple.  That's a little obtuse but I'm assuming this means that when we are processing one of these txns in `vm`, we should only set the code for that authority if the tuple is valid.  Is this method intended to serve that purpose or is this just a sanity check?

Reply:
No, it is used to verify that the authority list is properly formatted. It does not check the tuples, this is done in VM. For instance, it checks that the encoded fields are there, and they have no leading zeros in them.